### PR TITLE
Choose random host port in tests

### PR DESCRIPTION
**Pull request by mathbunnyru**
Related: https://github.com/jupyter/docker-stacks/issues/1622

### CODE REVIEW

File: base-notebook/test/test_container_options.py [hunk 1]
```diff
@@ -16,9 +16,11 @@ def test_cli_args(container: TrackedContainer, http_client: requests.Session) ->
     """Container should respect notebook server command line args
     (e.g., disabling token security)"""
     running_container = container.run_detached(
-        command=["start-notebook.sh", "--NotebookApp.token=''"]
+        command=["start-notebook.sh", "--NotebookApp.token=''"],
+        ports={"8888/tcp": None},
     )
-    resp = http_client.get("http://localhost:8888")
+    host_port = container.get_host_port("8888/tcp")
+    resp = http_client.get("http://localhost:" + host_port)
```
Comment (consideRatio) [inline base-notebook/test/test_container_options.py:23]:
```suggestion
    resp = http_client.get(f"http://localhost:{host_port}")
```

File: base-notebook/test/test_container_options.py [hunk 2]
```diff
@@ -35,13 +37,17 @@ def test_unsigned_ssl(
...
+        ports={"8888/tcp": None},
+    )
+    host_port = container.get_host_port("8888/tcp")
     # NOTE: The requests.Session backing the http_client fixture does not retry
     # properly while the server is booting up. An SSL handshake error seems to
     # abort the retry logic. Forcing a long sleep for the moment until I have
     # time to dig more.
     time.sleep(5)
-    resp = http_client.get("https://localhost:8888", verify=False)
+    resp = http_client.get("https://localhost:" + host_port, verify=False)
```
Comment (consideRatio) [inline base-notebook/test/test_container_options.py:50]:
```suggestion
    resp = http_client.get(f"https://localhost:{host_port}", verify=False)
```

File: base-notebook/test/test_start_container.py [hunk 3]
```diff
@@ -68,7 +69,8 @@ def test_start_notebook(
     assert len(expected_warnings) == len(warnings)
     # checking if the server is listening
     if expected_start:
-        resp = http_client.get("http://localhost:8888")
+        host_port = container.get_host_port("8888/tcp")
+        resp = http_client.get("http://localhost:" + host_port)
```
Comment (consideRatio) [inline base-notebook/test/test_start_container.py:73]:
```suggestion
        resp = http_client.get(f"http://localhost:{host_port}")
```

File: conftest.py [hunk 4]
```diff
@@ -108,6 +108,13 @@ def run_and_wait(
         assert rv == 0 or rv["StatusCode"] == 0
         return logs
 
+    def get_host_port(self, container_port: str) -> str:
+        assert isinstance(self.container, Container)
```
Comment (consideRatio) [inline conftest.py:112]: Some kind of docstring is always helpful!

```suggestion
        """Returns the host port associated with the tracked container's port."""
        assert isinstance(self.container, Container)
```

File: test/test_notebook.py [hunk 5]
```diff
@@ -10,7 +10,8 @@ def test_secured_server(
     container: TrackedContainer, http_client: requests.Session
 ) -> None:
     """Notebook server should eventually request user login."""
-    container.run_detached()
-    resp = http_client.get("http://localhost:8888")
+    container.run_detached(ports={"8888/tcp": None})
+    host_port = container.get_host_port("8888/tcp")
+    resp = http_client.get("http://localhost:" + host_port)
```
Comment (consideRatio) [inline test/test_notebook.py:15]:
```suggestion
    resp = http_client.get(f"http://localhost:{host_port}")
```

**Review by consideRatio — COMMENTED** [review]
LGTM!

Note for other reviewers and as a verification of my understanding by making it explicit: when we pass `ports={"8888/tcp": None}` to `run_detached`, it means that the container port 8888/tcp should be associated with the port `None` (random) in the host port.

https://github.com/jupyter/docker-stacks/blob/b020a0cf3b9616666b5ea286c3ed8af56e89c99e/conftest.py#L86-L89

```
    running_container = container.run_detached(
        command=["start-notebook.sh", "--NotebookApp.token=''"],
        ports={"8888/tcp": None},
    )
```

**Comment by mathbunnyru** [comment]
Thank you @consideRatio!

I want to confirm that after this change I am able to run `make test/base-notebook` in several tabs without any errors, which almost certainly means that we can run tests in parallel, both for the samge image or for different images as well.

**Comment by mathbunnyru** [comment]
Tests pass, so merging this :)

**Pull request merged**